Develop

Adam-DevPL opened the pull request:
Fix:
- delete Redis Service implementation (if stays Nest Cache method, if not I will make new implementation with ioredis package)
- tests for mongodb fails
- clean files structure

Features:
- new endpoint updateAnimal() - service + controller
- new endpoint createAnimal() - service + controller
- new endpoint addAnimalsList() - service + controller
- new endpoint addAnimalsListWithType() - service + controller
- tests for new endpoints

Adam-DevPL committed:
feat: endpoint getAnimal(id) + tests and dcoumentation

Adam-DevPL committed:
Merge branch 'develop' into get-single-animal

Adam-DevPL committed:
Merge pull request #10 from Adam-DevPL/get-single-animal

feat: endpoint getAnimal(id) + tests and dcoumentation

Adam-DevPL committed:
fix: some minor fixes for swagger docs and tests

Adam-DevPL committed:
feature: update animal, new endpoint in controller and method for service

Adam-DevPL committed:
test: make tests for update

Adam-DevPL committed:
gh_actions: modify deploy action

Adam-DevPL committed:
Merge pull request #11 from Adam-DevPL/update-animal

Update animal

Adam-DevPL committed:
gh_actions: more minor fixes

Adam-DevPL committed:
Merge pull request #13 from Adam-DevPL/update-animal

gh_actions: more minor fixes

Adam-DevPL committed:
fix: correct some github actions yml files

Adam-DevPL committed:
fix: next correction of github actions

Adam-DevPL committed:
fix: gh actions yml files

Adam-DevPL committed:
fix: small fix to gh actions

Adam-DevPL committed:
cleanup: gh actions

Adam-DevPL committed:
Merge pull request #14 from Adam-DevPL/update-animal

Update animal

Adam-DevPL committed:
feature: new endpoint createAnimal, service + controller

Adam-DevPL committed:
tests: new test for new endpoint createAnimal

Adam-DevPL committed:
feature: add new andpoint addAnimalsList - add list of animals to the db

Adam-DevPL committed:
tests and fix: add next endpoint - add list of animals and fix service

Adam-DevPL committed:
feature: new andpoint add list of animals with the same type

Adam-DevPL committed:
tests: for endpoint createing list of animals with one type

Adam-DevPL committed:
fix: test for service to crerate new Animal

Adam-DevPL committed:
fix: make safety adjustments

Adam-DevPL committed:
Merge pull request #15 from Adam-DevPL/create-animal

Create animal

Adam-DevPL committed:
Merge branch 'develop' into add-animals-list

Adam-DevPL committed:
Merge pull request #16 from Adam-DevPL/add-animals-list

Add animals list

Adam-DevPL committed:
Merge branch 'develop' into add-animals-one-type

Adam-DevPL committed:
Merge pull request #17 from Adam-DevPL/add-animals-one-type

Add animals one type

Adam-DevPL committed:
refactor: small cleaning and minor fixes

Adam-DevPL committed:
Merge pull request #18 from Adam-DevPL/develop-refactor

refactor: small cleaning and minor fixes

Parodix007 commented on src/animals/animals.service.ts:
```diff
@@ -1,15 +1,162 @@
+    try {
+      return await this.animalModel.find();
+    } catch (err) {
+      // console.error(err);
```
do wywalenia

Adam-DevPL replied:
zrobione

Parodix007 commented on src/animals/animals.service.ts:
```diff
@@ -1,15 +1,162 @@
 
-  async findAll() {
-    return this.animalModel.find();
+  async findAll(): Promise<Animal[]> {
```
Dto?

Adam-DevPL replied:
zrobione

Parodix007 commented on src/animals/animals.service.ts:
```diff
@@ -1,15 +1,162 @@
+      return await this.animalModel.find();
+    } catch (err) {
+      // console.error(err);
+      throw new InternalServerErrorException();
```
tutaj jakies customowe errory mozesz rzucac...

Adam-DevPL replied:
Okej, to tutaj jeszcze poprawię.

Parodix007 replied:
custoowy wyjatek. wywal console loga

Parodix007 commented on src/animals/animals.service.ts:
```diff
@@ -1,15 +1,162 @@
+        throw new BadRequestException('Animals already exist in database');
+      }
+
+      await this.redisCacheManager.clearCache();
```
czemu tak? nie chcemy usuwac tych ktore juz sa

Adam-DevPL replied:
Ja podszedłem do tego tak, że jak zachodzi jakaś zmiana w bazie danych to resetuje to co jest w Cache, aby przy nowym zapytaniu pobrać aktualną bazę, a nie to co jest w Cache (nieaktualne)

Adam-DevPL replied:
Poprawiłem i powinno być już dobrze :)

Parodix007 replied:
gdzie to masz wrzucone bo nie widze zmian

Adam-DevPL replied:
w RedisCache.service.ts mam zaimplementowaną metodę clearCache, która czyści zapytanie findAll i przy update zapytanie konkretnego zapytania findOne o podanym id

Adam-DevPL replied:
Podsumowując metoda clearCache z RedisCacheService, usuwa za każdym razem klucz findAll, przy zapytaniach POST i PATCH i dodatkowo usuwa klucz pojedynczego zapytania findOne dla danego id przy zapytaniu PATCH

Parodix007 commented on src/animals/dto/animal.dto.ts:
```diff
@@ -1,16 +1,95 @@
+
+export class UpdateAnimalDto extends PartialType(AnimalDto) {}
+
+// export class UpdateAnimalDto {
```
?

Adam-DevPL replied:
To jest taki sprytny mechanizm, że robi z AnimalDto kopię tylko, że ustawia wszystkie pola jako opcjonalne :)

Adam-DevPL commented on src/animals/animals.controller.ts:
```diff
@@ -1,30 +1,169 @@
+    description: 'Internal server error',
+    type: ErrorDto,
+  })
+  async getAnimal(@Param() { id }: AnimalIdParam): Promise<Animal> {
```
Zwraca Animal ze Schemy. A co dokładnie powinno zwracać?

Adam-DevPL replied:
zrobione

Adam-DevPL committed:
fix: make out dto

Adam-DevPL committed:
fix: tests fixed

Adam-DevPL committed:
fix: add swagger description for AniamlDtoResponse

Adam-DevPL committed:
Merge pull request #19 from Adam-DevPL/develop-refactor

fix: make out dto

Adam-DevPL committed:
fix: small improvment to redis cache

Adam-DevPL committed:
Merge pull request #20 from Adam-DevPL/develop-refactor

fix: small improvment to redis cache

Parodix007 commented on src/animals/__test__/animals.service.spec.ts:
```diff
@@ -40,5 +126,204 @@ describe('AnimalsService', () => {
+        throw new Error('mongodb down');
+      });
+
+      expect(service.findAll).rejects.toMatchSnapshot();
```
tutaj chcialbym abys sprawdzil czy dostajesz error domenowy w tym wypadku to podejscie nie jest takie przjrzyste

Adam-DevPL replied:
nie za bardzo rozumiem, co dokładnie mam zrobić??

Adam-DevPL replied:
zrobione

Parodix007 commented on src/animals/__test__/animals.service.spec.ts:
```diff
@@ -40,5 +126,204 @@ describe('AnimalsService', () => {
     });
+
+    it('should throw Internal Exception Error when something goes wrong with mongodb connection', async () => {
+      jest.spyOn(model, 'find').mockImplementation(() => {
```
w takich wypadach masz `//when`

Adam-DevPL replied:
poprawione

Adam-DevPL committed:
fix: tests describtion

Adam-DevPL committed:
Merge pull request #21 from Adam-DevPL/develop-refactor

fix: tests describtion

Adam-DevPL committed:
fix: improve falsy tests

Adam-DevPL committed:
Merge pull request #22 from Adam-DevPL/develop-refactor

fix: improve falsy tests

Parodix007 commented on src/animals/dto/animal.dto.ts:
```diff
@@ -1,16 +1,131 @@
   description?: string;
 }
+
+export class UpdateAnimalDto extends PartialType(AnimalDto) {}
```
jak to dokladnie dziala, jakie pola bedzie miec?

Adam-DevPL replied:
będzie mieć identyczne pola jak AnimalDto tylko wszystkie pola będą zaznaczone jako opcjonalne

Parodix007 commented on src/animals/dto/animal.dto.ts:
```diff
@@ -1,16 +1,131 @@
+  })
+  description?: string;
+
+  constructor(
```
czemu taki konstruktor?

Adam-DevPL replied:
fakt, tutaj mogę zrobić private, jeżeli o to chodzi :)

Parodix007 commented on src/redis-cache/redis-cache.service.ts:
```diff
@@ -1,4 +1,14 @@
+export class RedisCacheService {
+  constructor(@Inject(CACHE_MANAGER) private readonly cacheManager: Cache) {}
+
+  async clearCache(id?: string) {
```
2 oddzielne metody to powinny byc. Masz tutaj 2 rozne konteksty

Adam-DevPL replied:
w sensie nie mogę łączyć ze sobą operacji na redisie, które dotyczą różnych modułów lub metod?

Parodix007 replied:
no powinienes zawsze tworzyc dedykowane rzeczy a nie flagowac metody ifem... Wzorzec Strategia o tym mowi chociazby ;)

Adam-DevPL committed:
fix: next minor fixes:
- change access modifier for constructor in AnimalDtoResponse;
- seperate metohd for cache cleaning;
- clean comments

Adam-DevPL committed:
Merge pull request #23 from Adam-DevPL/develop-refactor

fix: next minor fixes:

Adam-DevPL committed:
Merge branch 'main' into develop

Parodix007 commented on src/animals/animals.service.ts:
```diff
@@ -1,15 +1,175 @@
+
+      return AnimalDtoResponse.mapperDto(animal);
+    } catch (err) {
+      if (err instanceof NotFoundException) {
```
takie cos ja bym widzial w filtrach ;)

Adam-DevPL replied:
Ogarnąłem :)
Zmieniłem trochę podejście. Zostawiłem obsługę błędów NestJs i filtrowi, który stworzyłem. 
Dodałem odpowiednie zmiany w testach, aby sprawdzić, czy faktycznie to działa i działa :)

Adam-DevPL committed:
fix: animal.service.ts and test for it

Adam-DevPL committed:
fix: security fix

Adam-DevPL committed:
fix: another security fix

Adam-DevPL committed:
fix: minor one

Adam-DevPL committed:
Merge pull request #24 from Adam-DevPL/develop-refactor

fix: animal.service.ts and test for it
